sympy/polys/rings.py

```diff
@@ -3005,3 +3114,234 @@ def sqf_list(f, all=False):
 
     def factor_list(f):
         return f.ring.dmp_factor_list(f)
+
+def cont_prim(p, x):
+    """Returns the content and primitive part of a polynomial with respect to a
+    specified variable."""
+
+    coeffs = p.coeff_split({x})
+    content = gcd_prs(coeffs)
+    primitive = p.exquo(content)
+    return content, primitive
+
+def monomial_extract(polynomials):
+    """
+    Extracts any common monomial from the polynomials in p.
+
+    Examples
+    ========
+
+    >>> from sympy.polys.rings import monomial_extract
+    >>> from sympy import ZZ, ring
+    >>> R, x, y = ring("x, y", ZZ)
+
+    >>> f = x + y
+    >>> g = x**2*y + x*y
+    >>> polynomials = [f, g]
+    >>> monomial_extract(polynomials)
+    ([x + y, x**2*y + x*y], None)
```

Comment:
I think it would be better to return `R.one` here rather than `None`.

Then the invariant can be demonstrated for the second example like:
```
    >>> f = x**2*y
    >>> g = x**2*y + x*y
    >>> monomial_extract([f, g])
    ([x, x + 1], x*y)
    >>> [fp, gp], gcdm = monomial_extract([f, g])
    >>> [fp*gcdm, gp*gcdm] == [f, g]
    True
```